integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -766,38 +418,54 @@ fn test_resharding_v3_base(params: TestReshardingParameters) {
 
         let clients =
             client_handles.iter().map(|handle| &test_loop_data.get(handle).client).collect_vec();
-        let client = clients[client_index];
 
+        // Skip if we already checked the latest height
         let tip = get_smallest_height_head(&clients);
+        if latest_block_height.get() == tip.height {
+            return false;
+        }
+        if latest_block_height.get() == 0 {
+            println!("State before resharding:");
+            print_and_assert_shard_accounts(&clients, &tip);
+        }
+        latest_block_height.set(tip.height);
 
         // Check that all chunks are included.
+        let client = clients[client_index];
         let block_header = client.chain.get_block_header(&tip.last_block_hash).unwrap();
-        if latest_block_height.get() < tip.height {
-            if latest_block_height.get() == 0 {
-                println!("State before resharding:");
-                print_and_assert_shard_accounts(&clients, &tip);
-            }
-            trie_sanity_check.assert_state_sanity(&clients, expected_num_shards);
-            latest_block_height.set(tip.height);
-            if params.all_chunks_expected && params.chunk_ranges_to_drop.is_empty() {
-                assert!(block_header.chunk_mask().iter().all(|chunk_bit| *chunk_bit));
-            }
+        if params.all_chunks_expected && params.chunk_ranges_to_drop.is_empty() {
+            assert!(block_header.chunk_mask().iter().all(|chunk_bit| *chunk_bit));
         }
 
-        // Return true if we passed an epoch with increased number of shards.
+        trie_sanity_check.assert_state_sanity(&clients, expected_num_shards);
+
         let epoch_height =
             client.epoch_manager.get_epoch_height_from_prev_block(&tip.prev_block_hash).unwrap();
-        assert!(epoch_height < 6);
-        let prev_epoch_id =
-            client.epoch_manager.get_prev_epoch_id_from_prev_block(&tip.prev_block_hash).unwrap();
-        let epoch_config = client.epoch_manager.get_epoch_config(&prev_epoch_id).unwrap();
-        if epoch_config.shard_layout.num_shards() != expected_num_shards {
-            return false;
+
+        // Return false if we have not yet passed an epoch with increased number of shards.
+        if new_layout_epoch_height.get().is_none() {
+            assert!(epoch_height < 6);
```

Comment:
is it correct to use `6` here? or is it better to use `params.epoch_length`?

Reply:
This one is epoch height, unrelated to epoch length.

Reply:
You are right, sorry!